test/resources/config/regexp.bal

```diff
@@ -0,0 +1,24 @@
+function name() {
+
+    string pattern = "ABC";
+    
+    string:RegExp _ = re `abc`;
+    string:RegExp _ = re `(abc)`;
+    string:RegExp _ = re `[abc]`;
+
+    string:RegExp _ = re `\p{gc=Ll}|\P{Lm}|\p{sc=Latin}`;
+    string:RegExp _ = re `\u{12FFFF}`;
+
+    string:RegExp _ = re `${pattern}`;
+    string:RegExp _ = re `\w\s\n`;
+
+    string:RegExp _ = re `(?-:abc)`;
+    string:RegExp _ = re `(?i-:abc)`;
+    string:RegExp _ = re `(?im-sx:abc)`;
+
+    string:RegExp _ = re `(abc)|${pattern}|[abc]|(?i:abc)`;
+    string:RegExp _ = re `[((abc))))\]]`;
```

Comment:
Better separate the negative syntax test cases. Can use a comment within this bal file itself. 
Also, check how this works for multi-lines, where the regexp is spanning across multiple lines.

Reply:
E.g.:

```
string:RegExp _ = re `This \n
is a \r\n
multi-line
comment
`;
```

Reply:
We are not adding negative tests.

Reply:
<img width="293" alt="Screenshot 2023-05-11 at 10 55 52" src="https://github.com/ballerina-platform/ballerina-grammar/assets/46857198/26168a69-3d1b-4100-92aa-7e8081386b20">

It's work for multi lines.